lib/migration.c

```diff
@@ -534,6 +534,60 @@ migration_region_access(vfu_ctx_t *vfu_ctx, char *buf, size_t count,
     return count;
 }
 
+bool
+migration_feature_supported(uint32_t flags) {
+    switch (flags & VFIO_DEVICE_FEATURE_MASK) {
+        case VFIO_DEVICE_FEATURE_MIGRATION:
+            return !(flags & VFIO_DEVICE_FEATURE_SET); // not supported for set
```

Comment:
and this should either be handled in migration_feature_set(), or the handler above.

Reply:
Does it not make sense to check support for the feature support all in one place? To me at least it wouldn't make sense to return `true` from `migration_feature_supported` in the case, for example, where `flags = VFIO_DEVICE_FEATURE_MIGRATION | VFIO_DEVICE_FEATURE_SET` - in this case, the feature is not supported for setting?

Reply:
This really is an implementation of "probe" if I understand things right. So you could have a "probe()" function in the general code, that delegates to migration.c for the migration flags (which of course are currently all of them!). Up to you.